File: script/sceneGraph/GraphNode.js
define(["sceneGraph/ModelViewMatrix"],
    function () {
        "use strict";

        /**
         * A node in the scene graph.
         * @constructor
         */
        var GraphNode = function () {
            /**
             *
             * @type {GraphNode[]}
             */
            this.children = [];
        };

        /**
         *
         * @param {GraphNode} node
         */
        GraphNode.prototype.append = function (node) {
            this.children.push(node);
        };

        return GraphNode;
    }
);

File: lib/scene_graph/graph_node.dart
part of scene_graph;

/**
 * A node in a scene graph. This is the abstract class scene graph nodes must
 * implement.
 */
abstract class GraphNode {
  List<GraphNode> children = new List();

  /**
   * Append a child node to this node.
   */

  void append(GraphNode child) {
    children.add(child);
  }
}